### PR TITLE
Bump datadog version to 0.65.0

### PR DESCRIPTION
**Pull request by clarkds14**
Updating datadog tracer version to 0.65.0 (latest): https://github.com/DataDog/dd-trace-java/releases/tag/v0.65.0.

Couple notable changes:
- We can no longer cast Span to DDSpan. See https://github.com/DataDog/dd-trace-java/releases/tag/v0.51.0.
- OpenTracing API changed such that the way to register the tracer at startup (in DatadogTracingBackendModule) changed. See https://github.com/DataDog/dd-trace-java/releases/tag/v0.51.0.
- The Writer interface in Datadog changed to include a new flush method. See https://github.com/DataDog/dd-trace-java/releases/tag/v0.65.0.

Note that the newer version has enhancements that fixes known bugs and usability issues observed in Misk traces/spans.

Verified these changes locally, the tracer was configured properly.

### CODE REVIEW

File: misk-gcp/src/main/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancer.kt [hunk 1]
```diff
@@ -18,8 +18,8 @@ class TracingLoggingEnhancer : LoggingEnhancer {
   fun enhanceLogEntry(tracer: Tracer, builder: LogEntry.Builder) {
     val activeSpan = tracer.activeSpan()
     var traceId: String? = null
-    when (activeSpan) {
-      is DDSpan -> traceId = activeSpan.context().toTraceId()
+    when (tracer) {
+      is DDTracer -> traceId = activeSpan.context().toTraceId()
```
Comment (mightyguava) [inline misk-gcp/src/main/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancer.kt:22]: hmm... if this doesn't depend on casting the Tracer, why do you even need this type switch?

File: misk-gcp/src/test/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancerTest.kt [hunk 2]
```diff
@@ -50,6 +50,10 @@ class NoopWriter : Writer {
   override fun close() {
   }
 
+  override fun flush() : Boolean {
+    return true
```
Comment (mightyguava) [inline misk-gcp/src/test/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancerTest.kt:54]: ?

File: misk-hibernate-testing/src/test/kotlin/misk/jdbc/SpanInjectorTest.kt [hunk 3]
```diff
@@ -43,7 +44,7 @@ class SpanInjectorTest {
       val query = "SELECT * FROM table"
       val transformInfo = TransformInfo(null, null, query, false, 0)
       val result = injector.transformQuery(transformInfo)
-      assertThat(result).contains((span as DDSpan).traceId.toString())
+      assertThat(result).contains((span as Span).context().toTraceId())
```
Comment (mightyguava) [inline misk-hibernate-testing/src/test/kotlin/misk/jdbc/SpanInjectorTest.kt:47]: do you need this cast?

File: misk-datadog/src/main/kotlin/misk/tracing/backends/datadog/DatadogTracingBackendModule.kt [hunk 4]
```diff
@@ -1,19 +1,22 @@
...
  */
 class DatadogTracingBackendModule : KAbstractModule() {
   override fun configure() {
-    // A DDTracer is installed by the DataDog Java agent, which runs before the app's main() method.
-    val tracer = GlobalTracer.get()
-    if (tracer is Tracer) {
+    // N.B. For information on the registration of the DDTracer w/ OpenTracing,
+    // see https://docs.datadoghq.com/tracing/custom_instrumentation/java/
+    val tracer = DDTracer.builder().build()
+    if (GlobalTracer.registerIfAbsent(tracer)) {
```
Comment (mightyguava) [inline misk-datadog/src/main/kotlin/misk/tracing/backends/datadog/DatadogTracingBackendModule.kt:18]: when does this return true/false?

**Review by mightyguava — APPROVED** [review]
LGTM but we might have to take some thought on how to roll this out safely. Looks like they broke API compatibility. But we can't guarantee that the tracer java agent is deployed with the same version as the misk library

File: misk-gcp/src/test/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancerTest.kt (hunk 2 above)
Comment (clarkds14) [reply misk-gcp/src/test/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancerTest.kt:54]: This is test-only code. See commit message. They added a flush method to their Writer interface.

File: misk-datadog/src/main/kotlin/misk/tracing/backends/datadog/DatadogTracingBackendModule.kt (hunk 4 above)
Comment (clarkds14) [reply misk-datadog/src/main/kotlin/misk/tracing/backends/datadog/DatadogTracingBackendModule.kt:18]: If there is already a tracer registered to the OpenTracing global tracer. Shouldn't happen unless someone else went and registered some other tracer.

File: misk-hibernate-testing/src/test/kotlin/misk/jdbc/SpanInjectorTest.kt (hunk 3 above)
Comment (clarkds14) [reply misk-hibernate-testing/src/test/kotlin/misk/jdbc/SpanInjectorTest.kt:47]: nope

File: misk-gcp/src/main/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancer.kt (hunk 1 above)
Comment (clarkds14) [reply misk-gcp/src/main/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancer.kt:22]: correct. cargo-culting is bad.

**Pull request merged**